internal/bzlmod/utils.bzl

```diff
@@ -119,6 +119,17 @@ def with_replaced_or_new_fields(_struct, **replacements):
 
     return struct(**new_struct_assignments)
 
+def repo_name(importpath):
+    path_segments = importpath.split("/")
+    segments = reversed(path_segments[0].split(".")) + path_segments[1:]
+    candidate_name = "_".join(segments).replace("-", "_")
+
+    def _encode_case(c):
```

Comment:
changing the encoding isn't a bad idea, but I do worry that this encoding itself could cause more conflicts. Is there a more conflict resistant encoding? Also, I believe any encoding change would require both the extension and gazelle BUILD.bazel generation to be in proper sync.

Reply:
I'm not super convinced yet [that supportive case sensitivity is the right call](https://github.com/bazelbuild/bazel-gazelle/pull/1949/files#r1793727670) – obviously open to convincing, but sharing my current gut-feeling.

Reply:
! isn't valid in repo names though.